solver/src/driver.rs

```diff
@@ -334,9 +336,13 @@ impl Driver {
             .await
             .context("failed to get current block")?
             .as_u64();
+
+        let inflight_trades = &self.inflight_trades.clone();
+        // Reset the inflight orders - event if get_liquidity fails.
+        self.inflight_trades = HashSet::new();
```

Comment:
One cool trick you can use to do this more succinctly is:
```suggestion
        let inflight_trades = std::mem::replace(&mut self.inflight_trades, HashSet::new());
```

Which would avoid the `clone`.

Also, I'd like to bring back the point (https://github.com/gnosis/gp-v2-services/pull/662#discussion_r642385489) that I don't think we ever need to clear the inflight trades early. Specifically I think we should just ever clear this on a successful run. So, from line `407`:
```rust
if let Some(mut settlement) = rated_settlements.clone().into_iter().max_by(...) {
    ...
    self.inflight_trades = ...
} else {
    self.inflight_trades = HashSet::new();
}
```
https://github.com/gnosis/gp-v2-services/blob/c368cd97c1d720fd8c508a3c80d29353cdc1e49d/solver/src/driver.rs#L424-L430

With the way it is currently, imagine a situation where:
1. The inflight trades set gets prematurely cleared
2. There is an intermittent network error that causes the fetching the orders to fail
3. Run loops gets restarted right away
4. Orders are fetched containing inflight orders that don't get filtered.

Additionally, I don't even think its a big deal to never clear this - the worse that can happen is the inflight trades become old (but there aren't any new ones), so it would just try to filter out orders that the orderbook API already knows were filled.

Reply:
Given all these options I would go for the option of not clearing it, but only updating it when necessary. Will wait for @fleupold's input here.

Reply:
The arguments make sense. I can get behind not clearing or only replacing on successful submission.

Reply:
I will go with that then!